From a824826fc6daedf8b1308bfc626540a9e64d05f8 Mon Sep 17 00:00:00 2001
From: Yumi Yukimura <me.cafebabe@gmail.com>
Date: Tue, 2 Jan 2024 07:34:14 +0800
Subject: [PATCH] init: devices: Add option to accept any device as boot device

Targets like PCs/VMs usually don't have fixed boot devices.

For example, moving the disk to another controller would result
in getting different block device path.

To workaround this situation, add an option to bypass the boot
device filter, so that every block devices would get their
symlink in /dev/block/by-name/.

Usage: Specify `androidboot.boot_devices=any` on bootconfig or cmdline

Test: /dev/block/by-name/* exists when `androidboot.boot_devices=any`
Change-Id: I5ae0faf401dac71352dbb05115a2dd6f3e76adc2
---
 init/devices.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/init/devices.cpp b/init/devices.cpp
index cead726167..e72a504dfe 100644
--- a/init/devices.cpp
+++ b/init/devices.cpp
@@ -223,7 +223,8 @@ BlockDeviceInfo DeviceHandler::GetBlockDeviceInfo(const std::string& uevent_path
         info.str = "";
     }
 
-    info.is_boot_device = boot_devices_.find(info.str) != boot_devices_.end();
+    info.is_boot_device = boot_devices_.find(info.str) != boot_devices_.end() ||
+        boot_devices_.find("any") != boot_devices_.end();
 
     return info;
 }
-- 
2.39.5

